src/d3a/models/market/market_structures.py

```diff
@@ -20,20 +20,22 @@
 
 
 class Offer:
-    def __init__(self, id, price, energy, seller, original_offer_price=None):
+    def __init__(self, id, price, energy, seller, original_offer_price=None, energy_origin=None):
         self.id = str(id)
         self.real_id = id
         self.price = price
         self.original_offer_price = original_offer_price
         self.energy = energy
         self.seller = seller
+        self.energy_origin = energy_origin
 
     def __repr__(self):
         return "<Offer('{s.id!s:.6s}', '{s.energy} kWh@{s.price}', '{s.seller} {rate}'>"\
             .format(s=self, rate=self.price / self.energy)
 
     def __str__(self):
-        return "{{{s.id!s:.6s}}} [{s.seller}]: {s.energy} kWh @ {s.price} @ {rate}"\
+        return "{{{s.id!s:.6s}}} [ORIGIN: {s.energy_origin}] " \
```

Comment:
Can you please convert origin to lowercase? IMO uppercase words in logs are not ideal.